services/api/src/utils/middleware/authenticate.js

```diff
@@ -60,7 +65,7 @@ async function fetchUser(ctx, next) {
     const { User } = mongoose.models;
     const user = await User.findById(jwt.sub);
     if (!user || (jwt.jti && jwt.jti !== user.authTokenId)) {
-      ctx.throw(401, 'user associated to token could not be found');
+      throw new TokenError('user associated to token could not be found');
```

Comment:
I like this error to be displayed to the user, as this often is our fault, the only time i have seen it in the past is because we loaded the wrong fixtures or flushed the db. Happen quite a bit in acrylic. 

I guess its a bit of edge case, but i am still wondering if there is something we can do to bring this error to our attention. 

also i think it would be good split off `jwt.jti !== user.authTokenId ` into another error - so the case where the user is not found and jti not match are 2 different errors.

Reply:
it happens all the time during dev as we are reloading data. I don't know how often it happens in a production environment, we would have to know why but the point is that it isn't recoverable and the user must be logged out.

> also i think it would be good split off jwt.jti !== user.authTokenId into another error - so the case where the user is not found and jti not match are 2 different errors.

That might make sense...what different action would/could they take based on this however? Perhaps if the user has been deleted? We would have to think about it.